### PR TITLE
Driver error enum

### PR DESCRIPTION
**Pull request by Wenzel**
This PR adds driver specific error types based on `thiserror` for Xen and KVM

regarding virtualbox, the FDP crate returns `Box<dyn Error>` for every function, so implementing a better handling for this driver will start with FDP library.

### CODE REVIEW

File: src/driver/xen.rs [hunk 1]
```diff
@@ -29,6 +33,24 @@ pub struct Xen {
     back_ring: vm_event_back_ring,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum XenDriverError {
+    #[error("event version mismatch: {0} <-> {1}")]
+    EventVersionMismatch(u32, u32),
+    #[error(transparent)]
```
Comment (rageagainsthepc) [inline src/driver/xen.rs:40]: IIRC this duplicates the error message from the source error and therefore makes walking error source chains awkward.

Comment (Wenzel) [reply src/driver/xen.rs:40]: So we should replace `#[error(transparent)]` by `#[error("defined error message")]` ?

Comment (Wenzel) [reply src/driver/xen.rs]: okay, done

**Comment by Wenzel** [comment]
Merging this to work on #151

**Pull request merged**